src/utils/channel_utils.js

```diff
@@ -656,3 +656,15 @@ export function filterChannelsMatchingTerm(channels: Array<Channel>, term: strin
             displayName.startsWith(lowercasedTerm);
     });
 }
+
+export function filterGroupChannelsMatchingTerm(groupChannels: Array<Channel>, term: string): Array<Channel> {
+    const lowercasedTerm = term.toLowerCase();
+
+    return groupChannels.filter((channel: Channel): boolean => {
+        const displayName = (channel.display_name || '').toLowerCase();
```

Comment:
Unfortunately I'm not sure if this will work in all cases. Mainly if the group channel includes a lot of users or users with longer usernames since we truncate the display_name here if it exceeds the character limit https://github.com/mattermost/mattermost-server/blob/master/model/channel.go#L228

I can't think of a good way around that without adding the other server endpoint for searching groups or by including some more metadata in the GMs when we create them (which causes problems with migration)

Happy to hop on a call if you want to discuss